src/main/kotlin/org/rust/ide/utils/RustModTabTitleProvider.kt

```diff
@@ -0,0 +1,18 @@
+package org.rust.ide.utils
+
+import com.intellij.openapi.fileEditor.impl.EditorTabTitleProvider
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.vfs.VirtualFile
+import org.rust.lang.core.psi.impl.isRustFile
+
+class RustModTabTitleProvider : EditorTabTitleProvider {
+    val MOD_FILE_NAME = "mod.rs"
+
+    override fun getEditorTabTitle(project: Project, file: VirtualFile): String? {
+        if (file.isRustFile && MOD_FILE_NAME == file.name) {
+            return "${file.parent.name}/${file.name}"
```

Comment:
@mbStavola yep, this is what I meant, but looks like I was wrong about the details :) 

TabTitleProvider should return `null` when it is not applicable, and the defalt name should be used. So, UniqueNameProvider will return `null` when a name will be unique, and this will cause this provider to return `null` as well. 

I *think* the proper logic should be along this lines:

```
if is not RustFile: return null
if name != "mod.rs" return null
val unique = super.getEditorTabTitle()
if (unique != null) return unique
return "${file.parent.name}/${file.name}"
```

But maybe I am missing something again? Better to verify it!